app/models/framework/definition/parser.rb

```diff
@@ -24,9 +24,9 @@ class Parser < Parslet::Parser
       rule(:field_def)            { unknown_field | known_field | additional_field }
       rule(:known_field)          { optional >> additional_field_identifier.absent? >> pascal_case_identifier.as(:field) >> from_specifier }
       rule(:additional_field)     { optional >> type >> space >> additional_field_identifier.as(:field) >> from_specifier }
-      rule(:unknown_field)        { optional >> primitive_type >> space >> from_specifier }
-      rule(:type)                 { pascal_case_identifier.as(:type) }
-      rule(:primitive_type)       { str('String').as(:type) }
+      rule(:unknown_field)        { optional >> primitive_type.as(:type) >> space >> from_specifier }
+      rule(:type)                 { (primitive_type | pascal_case_identifier.as(:lookup)).as(:type) }
```

Comment:
This is an idiom we're using a lot. I'm not sure I'd want to effectively inline this everywhere else. The alternative is to split over lines in a `do` rule block but that makes the grammar much taller. Also, since `type` has subsequently been renamed to `type_def` to enforce a semantic separation between a type and its definition, this would make the line a lot longer.

I'd prefer to leave this concise; would understand if you'd rather not, but then to satisfy rubocop we'd have to start breaking the rule up over three lines with a `do`.  I don't think there are any easy wins here.